cmake/common/eprosima_libraries.cmake

```diff
@@ -60,40 +102,95 @@ macro(eprosima_find_package package)
                 if(NOT IS_TOP_LEVEL)
                     set(${package}_FOUND TRUE PARENT_SCOPE)
                 endif()
+                message(STATUS "Found ${package}: ${PROJECT_SOURCE_DIR}/thirdparty/${package}")
             endif()
         endif()
 
-        if(${package}_FOUND)
-            message(STATUS "${package} library found...")
-        elseif(${FIND_REQUIRED})
-            message(FATAL_ERROR "${package} library not found...")
-        else()
-            message(STATUS "${package} library not found...")
+        # If the package was found we have already show an status message.
+        # if it was not found and it was required throw an fatal error.
+        # if if was not found but it was not required show a warning.
+        if((NOT ${package}_FOUND) AND ${FIND_REQUIRED})
+            message(FATAL_ERROR "${package} package NOT found")
+        elseif(NOT ${package}_FOUND)
+            message(STATUS "${package} package NOT found")
         endif()
     endif()
 endmacro()
 
+# Macro to find all Fast DDS thirdparty libraries expect for Fast CDR (look at eprosima_find_package).
+#
+# Arguments:
+#   :package: The name of the packge to find. Used for find_package(${package})
+#   :thirdparty_name: The name of the package directory under thirdparty, i.e. thirdparty/${thirdparty_name}
+#
+# Related CMake options:
+#   :THIRDPARTY: Activate the use of internal thirdparties if external packges is not found [Defaults: OFF]
+#   :THIRDPARTY_FORCE: Force the use of internal thirdparties [Defaults: OFF]. Set to ON if EPROSIMA_BUILD is set to ON.
```

Comment:
Looking at the code below, I see that `EPROSIMA_BUILD` sets `THIRDPARTY` to ON, not `THIRDPARTY_FORCE`. Either change the behavior or the documentation.